atlasdb-perf/src/test/java/com/palantir/atlasdb/performance/backend/KeyValueServiceInstrumentationTest.java

```diff
@@ -33,6 +33,8 @@ public void forDatabaseAddsInstrumentationFromCorrectClassName() {
                 .isExactlyInstanceOf(MockKeyValueServiceInstrumentation.class);
         assertThat(KeyValueServiceInstrumentation.forDatabase(mockKeyValueServiceInstrumentation.toString()))
                 .isExactlyInstanceOf(MockKeyValueServiceInstrumentation.class);
+
+        KeyValueServiceInstrumentation.removeBackendType(mockKeyValueServiceInstrumentation);
```

Comment:
It's a bit convoluted, but `forDatabase` does eventually call `addNewBackendType` if the argument is a backend that is not yet mapped